apps/next/.env.example

```diff
@@ -22,4 +22,7 @@ POSTGRES_CA_DATA=x
 POSTGRES_HOST=x
 POSTGRES_USER=x
 POSTGRES_PASSWORD=x
-POSTGRES_DATABASE=x
\ No newline at end of file
+POSTGRES_DATABASE=x
+CLOUD_BUILDER_NAME=x
+DOCKERHUB_ORG=x
+DOCKERHUB_REPO=x
```

Comment:
also probably makes sense to use a tmp directory for docker config, and to use it with all docker commands e.g. `docker --config <> login`, that way 1) the shared system login state isn't messed with and 2) concurrent requests aren't contending for the same config file (this might be important if we let users authenticate with their own docker credentials to push to a private registry)

(docker login generates a config file with a token, and that config file can be anywhere but defaults to ~/.docker/config.json)

Reply:
Yes I did explore storing it in the shell env and using the Docker token as an API token, but I couldn't find docs on what to name the env variable in order to match what Docker expects. I'll search again to see if I find any docs, but here's what Docker's CLI shows:
<img width="612" alt="Screenshot 2024-05-15 at 18 22 02" src="https://github.com/onmetal-dev/metal/assets/10248833/8934ef87-8d20-48f1-a74c-612e440b57f4">

Reply:
I just logged the contents of my `~/.docker/config.json` and it didn't contain a token, even after I re-logged in.

I would say that doing `docker login` for every/most requests will probably be inefficient. We can login once during our container's startup.

Reply:
Apart from that, I can do a login if the user provides credentials, and those creds will be used for all subsequent commands. LMK what you think.

Reply:
if you have docker desktop running then you won't see a token

I just tried `docker login -u <> --password <>` on a fresh ubuntu:22.04 container that just has the docker client installed and .docker/config.json has 

```
{
        "auths": {
                "https://index.docker.io/v1/": {
                        "auth": "<>"
                }
        }
}
```

Also it turns out the "auth" field is just the base64 encoded `username:password` (https://stackoverflow.com/a/45671103/205138). 

So putting that all together I think it makes sense for the logic here to create a tmp dir for the request, populate a config.json file, and then run all commands from that directory, e.g. `docker --config config.json ...`

Reply:
Roger that. Note though that I suspect that we won't be able to get `.docker/config.json` from the uploaded tarball, as people won't (or shouldn't?) use git to track a file with such credentials. Instead, I propose using flags of the "up" command to specify the said values e.g
```
metal up --docker-registry <> --docker-username <> --docker-password <>
```

I will also try to enable reading the values from env vars like `METAL_DOCKER_REGISTRY` so as to save the user from having to specify the values for each command, or even specifying the value on the command line at all.

Reply:
don't worry about capturing user or upload-specific docker credentials. I think this will be configured elsewhere. I just want the server-side logic to be flexible enough to accommodate variable credentials

Reply:
Noted. Please take a look at this and LMK what you think:
https://github.com/onmetal-dev/metal/pull/21/commits/bc40ce29be6293d3a90546dbd233973035627ad0